pwiz_tools/Skyline/Util/UtilIO.cs

```diff
@@ -954,7 +954,10 @@ public static void SafeDelete(string path)
         {
             try
             {
-                Helpers.TryTwice(() => Directory.Delete(path, true), $@"Directory.Delete({path})");
+                if (path != null && Directory.Exists(path)) // Don't waste time trying to delete something that's already deleted

```

Comment:
Surprising that this is meaningful, but it sounds like you have testing proof.

Reply:
Those "directory does not exist" exceptions slow things down, especially under code coverage where we allot extra retry time. A similar check already exists for FileEx.SafeDelete.

Reply:
Good point. Throwing exceptions is slow, especially when we are going to try multiple times. Makes you wonder if we should have a way to exclude certain exception times from the try again loop.